dipy/tracking/local/localtracking.py

```diff
@@ -93,12 +100,16 @@ def _generate_streamlines(self):
             directions = dg.initial_direction(s)
             directions = directions[:max_cross]
             for first_step in directions:
-                stepsF = local_tracker(dg, tc, s, first_step, vs, F, ss, fixed)
-                if stepsF < 0:
+                stepsF, tissue_class = local_tracker(dg, tc, s, first_step, vs, F, ss, fixed)
+                if not (self.return_all or
+                        tissue_class == 2 or  # ENDPOINT
```

Comment:
It looks like they added this feature pretty recently, so we'd need to bump
the minimum cython version to something like .21.
https://github.com/cython/cython/commit/50133b5a91eea348eddaaad22a606a7fa1c7c457

Workarounds discussed here,
https://groups.google.com/forum/#!topic/cython-users/gn1p6znjoyE

Bago

On Wed, Jan 28, 2015 at 2:51 PM, Gabriel Girard <notifications@github.com>
wrote:

> In dipy/tracking/local/localtracking.py
> <https://github.com/nipy/dipy/pull/565#discussion_r23731539>:
>
> > @@ -93,12 +100,16 @@ def _generate_streamlines(self):
> >              directions = dg.initial_direction(s)
> >              directions = directions[:max_cross]
> >              for first_step in directions:
> > -                stepsF = local_tracker(dg, tc, s, first_step, vs, F, ss, fixed)
> > -                if stepsF < 0:
> > +                stepsF, tissue_class = local_tracker(dg, tc, s, first_step, vs, F, ss, fixed)
> > +                if not (self.return_all or
> > +                        tissue_class == 2 or  # ENDPOINT
>
> Is there a way to access cdef enum TissueClass`from here? (defined in
> dipy.tracking.local.tissue_classifier.pyd)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nipy/dipy/pull/565/files#r23731539>.
>